platform/pwl.c

```diff
@@ -1103,6 +1111,53 @@ zxdg_shell_v6_ping (void                 *data G_GNUC_UNUSED,
 #endif /* HAVE_XDG_SHELL_UNSTABLE_V6 */
 
 
+static void
+xkb_data_init(PwlSeat *pwl_seat)
```

Comment:
You may want to rename this as `pwl_seat_create()`, and move allocation here, like this:

```c
static PwlSeat*
pwl_seat_create(struct wl_seat *seat)
{
    PwlSeat *self = /* allocate */;

    self->seat = seat;
    /* initialize more values */
    /* initialize the XKB values, too */
    /* attach the seat listener */
    return self;
}
```

Then later in `registry_global()` you only need to call `pwl_seat_create()` and add the new value to the list of seats.

Reply:
Done.